plugins/SimpleNtupler_miniAOD.cc

```diff
@@ -1318,27 +1323,31 @@ void SimpleNtupler_miniAOD::analyze(const edm::Event& event, const edm::EventSet
   int nJets = 0;
 
   for (std::vector<pat::Jet>::const_iterator itJet = jets->begin(); itJet != jets->end(); itJet++) {
-        if ((itJet->pt() > 30 && deltaR((*itJet),dil.daughter(0)->p4()) > 0.4 && deltaR((*itJet),dil.daughter(1)->p4()) > 0.4) && (((itJet->neutralHadronEnergyFraction() < 0.99 && itJet->neutralEmEnergyFraction() < 0.99 && (itJet->chargedMultiplicity()+itJet->neutralMultiplicity()) > 1 && itJet->muonEnergyFraction() < 0.8) && ((fabs(itJet->eta()) <= 2.4 && itJet->chargedHadronEnergyFraction() > 0 && itJet->chargedMultiplicity() > 0 && itJet->chargedEmEnergyFraction() < 0.99) || fabs(itJet->eta()) > 2.4) && fabs(itJet->eta()) <= 2.7) || (itJet->neutralHadronEnergyFraction() < 0.98 && itJet->neutralEmEnergyFraction() > 0.01 && itJet->neutralMultiplicity() > 2 && fabs(itJet->eta()) > 2.7 && fabs(itJet->eta()) <= 3.0) || (itJet->neutralEmEnergyFraction() < 0.90 && itJet->neutralMultiplicity() > 10 && fabs(itJet->eta()) > 3.0)) ) {
-           	if (nJets < 10){
-			t.jet_pt[nJets] = itJet->pt();		
-			t.jet_eta[nJets] = itJet->eta();		
-			t.jet_phi[nJets] = itJet->phi();
-			//t.jet_btag[nJets] = itJet->bDiscriminator("combinedSecondaryVertexV2BJetTags");		
-			t.jet_btag[nJets] = itJet->bDiscriminator("pfCombinedInclusiveSecondaryVertexV2BJetTags");
-		}
-	   	nJets++; 		
-	}
+      if ((itJet->pt() > 30 && deltaR((*itJet),dil.daughter(0)->p4()) > 0.4 && deltaR((*itJet),dil.daughter(1)->p4()) > 0.4) && (((itJet->neutralHadronEnergyFraction() < 0.99 && itJet->neutralEmEnergyFraction() < 0.99 && (itJet->chargedMultiplicity()+itJet->neutralMultiplicity()) > 1 && itJet->muonEnergyFraction() < 0.8) && ((fabs(itJet->eta()) <= 2.4 && itJet->chargedHadronEnergyFraction() > 0 && itJet->chargedMultiplicity() > 0 && itJet->chargedEmEnergyFraction() < 0.99) || fabs(itJet->eta()) > 2.4) && fabs(itJet->eta()) <= 2.7) || (itJet->neutralHadronEnergyFraction() < 0.98 && itJet->neutralEmEnergyFraction() > 0.01 && itJet->neutralMultiplicity() > 2 && fabs(itJet->eta()) > 2.7 && fabs(itJet->eta()) <= 3.0) || (itJet->neutralEmEnergyFraction() < 0.90 && itJet->neutralMultiplicity() > 10 && fabs(itJet->eta()) > 3.0)) ) {
```

Comment:
The jet threshold will have to go down to 20, at least for b-tagged jets.

Reply:
While it looks fine to me to change the 30GeV to 20GeV in the code at the moment, in future I would prefer to have a file defining our selection for histogramming. It is quite easy to implement and we won't have to change and recompile the actual code each time we want to tweak slightly the selection. 

another important thing: the code should evolve iteratively, so each pull request (PR) should be closed reasonably fast and only critical changes has to be implemented immediately. Everything else should deserve a separate github issue and be closed by another PR.

Reply:
I'm trying to fix the mumu.C file (ntuple running code). 
But the fixing if/else statements (applying Denis's comments to code) takes more time than I thought. 
(I'm looking for ways to simplify the code.)

Misha, your comment means that I have to close current PR before start another PR?
Or do you mean that I have to close the PR right after checking your comments?

Reply:
@hyeahyun my comment means we should **merge** it quicker, not **close**. And yes, the merging should happen before you start to work on another feature. So it is rather a comment for @rathjd then for you

Reply:
Thank you for your quick answer.
So... I'll apply Denis's comments as soon as possible for quick merging, and will push the code again.